src/test/compile-fail/not-panic-safe-2.rs

```diff
@@ -0,0 +1,22 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![allow(dead_code)]
+#![feature(recover)]
+
+use std::panic::PanicSafe;
+use std::rc::Rc;
+
+fn assert<T: PanicSafe + ?Sized>() {}
+
+fn main() {
+    assert::<Rc<i32>>(); //~ ERROR: is not implemented
```

Comment:
Yeah I was a little torn on this. On one hand `Rc<i32>` is panic safe, but `Rc<RefCell<i32>>` is not. The latter case isn't caught by the impls today, but perhaps I can use `Shared` (like we're using `Unique` now) to catch this case.